src/main/flight/position.c

```diff
@@ -22,205 +22,202 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <math.h>
-#include <limits.h>
 
 #include "platform.h"
 
+#if defined(USE_BARO) || defined(USE_GPS)
+
 #include "build/debug.h"
 
 #include "common/maths.h"
 #include "common/filter.h"
 
 #include "fc/runtime_config.h"
 
-#include "flight/position.h"
-#include "flight/imu.h"
-#include "flight/pid.h"
-
 #include "io/gps.h"
 
-#include "scheduler/scheduler.h"
-
 #include "sensors/sensors.h"
 #include "sensors/barometer.h"
 
-#include "pg/pg.h"
-#include "pg/pg_ids.h"
+#include "position.h"
 
+#ifdef USE_GPS
+static pt2Filter_t gpsUpsampleLpf;
+#endif
+#ifdef USE_BARO
+static pt2Filter_t baroUpsampleLpf;
+#endif
+
+static float altitudeCm = 0.0f;
```

Comment:
Would be nice but that's probably better left for a separate PR. I left the units alone because it was hard enough to refactor the cluttered code while simultaneously mixing in my changes.

Or maybe I can change the units to SI as a final commit at the end, after I checked that everything works as intended.